esp/templates/program/modules/resourcemodule/classrooms.html

```diff
@@ -74,3 +97,71 @@ <h2>Step 3: Classrooms</h2>
+<script src="https://rawgit.com/elo80ka/django-dynamic-formset/f5f68c818953ddf43ffa7cb2e9aa391c3d4dfb47/src/jquery.formset.js"></script>
+<!--jQuery.initialize plugin is created to help maintain dynamically created elements on the page-->
+<script src="https://rawgit.com/pie6k/jquery.initialize/master/jquery.initialize.js"></script>
+<script>
+<!--Formset settings-->
+$j('.furnishing-formset').formset({
+    prefix: 'furnishings',
+    formTemplate: '#id_empty_form',
+    addText: 'add furnishing',
+    deleteText: 'remove',
+    addCssClass: 'furnishing-add',
+    deleteCssClass: 'furnishing-delete',
+    formCssClass: 'furnishing-dynamic-form',
+});
+
+<!--Get choices for a furnishing via POST-->
+function getChoices(furnishing, callback){
+    var data = {furnishing: furnishing, csrfmiddlewaretoken: csrf_token()};
+    $j.post('/manage/{{ prog.url }}/ajaxfurnishingchoices', data, "json").success(callback);
+}
+
+<!--Update the option field to reflect whether there are specific choices (dropdown) or not (open response). If a value is supplied, set that value as selected afterwards.-->
+function update_choices(obj, value) {
+    var furnishing = $j(obj).val()
+    getChoices(furnishing, function(response) {
+        var choices = response.choices;
+        var num = $j(obj).attr('id').split("-")[1];
+        $j('#id_furnishings-' + num + '-choice').remove();
+        if (choices.length > 1) {
+            $j(obj).after('<select style="margin-left: 4px" id="id_furnishings-' + num + '-choice" name="furnishings-' + num + '-choice" type="text"></select>');
+            $j('#id_furnishings-' + num + '-choice').append('<option>(option)</option>');
+            for (i in choices) {
+                $j('#id_furnishings-' + num + '-choice').append('<option value=' + choices[i] + '>' + choices[i] + '</option>');
+            }
+        } else {
+            $j(obj).after('<input style="margin-left: 4px" id="id_furnishings-' + num + '-choice" name="furnishings-' + num + '-choice" maxlength="50" placeholder="(option)" type="text"></select>');
+        }
+        if (value) {
+            $j('#id_furnishings-' + num + '-choice').val(value);
+        }
+    });
+}
+
+<!--Upon page load and dynamic creation of new selects-->
+$j.initialize("select", function() {
+    <!--Selects start with (option) or (furnishing) by default, but we want to hide these as options in the dropdown.-->
+    var opt = $j(this).find("option")[0];
+    $j(opt).hide();
+    $j(opt).css("display", "none");
+    opt.disabled = true;
+    <!--Convert option field to dropdown if necessary-->
+    if ($j(this).attr('id').split("-")[2] == "furnishing") {
+        var num = $j(this).attr('id').split("-")[1];
+        <!--Make sure to preserve selected value-->
+        var value = $j('#id_furnishings-' + num + '-choice').val();
+        if (value) {
+            update_choices($j(this), value);
+        }
+    }
+}, { target: $j("#furnishing_formset")[0] });
```

Comment:
Why the `[0]`?  I think jquery doesn't actually require it here, although I could be wrong.

Reply:
I think this had something to do with the function requiring a single object, not a jQuery query. I've changed it anyways such that it just uses JS to get the object.